fixes #6129 feat(nimbus): Get control branch from metadata config, display in alert

LZoog opened the pull request:
fixes #6129, and will fix the bug described in #6120

Because:
* The control/reference branch, along with other properties that will be accounted for in a separate PR, can be overridden in the Jetstream config for that experiment which affects only the analysis and Results page

This commit:
* Adds a getControlBranchName fn to return the reference branch from the Jetstream metadata if it exists, or returns the control branch from analysis data
* Adds the new external_config property to the metadata for tests/Storybook and fills in the reference_branch property
* Displays an alert for users on the Results page if any overrides exist

---

Note: I'd like to test this locally with the experiment noted in #6120. Will do before merging (if you see this comment I haven't done that yet - ~~looks like we need the metadata reuploaded, will ping relevant folks~~ pinged Anna S. and she reuploaded it). Our "viewing analysis data locally" docs for any prod experiment are out of date, I may put in another PR up for doc updates there...

LZoog commented:
Ah oops, I think there's one thing I missed that's causing failures, will fix that real quick and reopen as PR when CI is green.

LZoog commented on app/experimenter/nimbus-ui/src/components/PageResults/ExternalConfigAlert/index.test.tsx:
```diff
@@ -0,0 +1,29 @@
+      <ExternalConfigAlert
+        externalConfig={{
+          ...MOCK_METADATA_EXTERNAL_CONFIG,
+          reference_branch: "foo",
```
I only changed this because I wanted to check for text context other than "control" in case we change "Baseline branch" text to use the word "control" later.

LZoog commented on app/experimenter/nimbus-ui/src/components/PageResults/TableVisualizationRow/index.stories.tsx:
```diff
@@ -29,6 +29,7 @@ storiesOf("pages/Results/TableVisualizationRow", module)
       metricKey="retained"
       displayType={DISPLAY_TYPE.POPULATION}
       group={GROUP.OTHER}
+      isControlBranch
```
I could have given this a default variable to make Storybook/maybe some future tests more convenient, but `isControlBranch` needs to be explicitely passed in everywhere we use it in the tables, so I'd rather keep it as required without a fallback.

lmorchard commented on app/experimenter/nimbus-ui/src/components/PageResults/ExternalConfigAlert/index.tsx:
```diff
@@ -0,0 +1,46 @@
+        click here
+      </LinkExternal>{" "}
+      to view the config file):
+      {externalConfig.reference_branch && (
```
Will there be any other external config properties mentioned here besides `reference_branch`? The `MetadataExternalConfig` type also includes these:
```
  end_date: string | null;
  enrollment_period: string | null;
  skip: boolean;
  start_date: string | null;
```

LZoog replied:
This PR is just for `reference_branch`, I noted the separate PR bit in the PR description / issue number in [a comment here](https://github.com/mozilla/experimenter/pull/6264/files#diff-1255a403cd47ad0be32d2918921c64471bdcf96d6808ffee9487be30fa0e815fR50) 😁

LZoog commented:
Pushed one tiny update, changed `reference_branch: "control"` in the mock config to `reference_branch: "treatment"` and added 1 other test since it was overriding but wasn't actually pointing to a different branch than what was already set as the reference branch.

lmorchard commented on app/experimenter/nimbus-ui/src/lib/visualization/types.ts:
```diff
@@ -19,6 +19,16 @@ export type AnalysisDataWeekly = Exclude<AnalysisData["weekly"], null>;
 export interface Metadata {
   metrics: { [metric: string]: MetadataPoint };
   outcomes: { [outcome: string]: MetadataPoint };
+  external_config?: MetadataExternalConfig | null;
+}
+
+export interface MetadataExternalConfig {
+  end_date: string | null;
+  enrollment_period: string | null;
+  reference_branch: string | null;
+  skip: boolean;
+  start_date: string | null;
+  url: string;
 }
```
Maybe just a note to myself, but: I was a little confused looking at [this change to the Jetstream schema](https://github.com/mozilla/jetstream/pull/841) and [this comment](https://github.com/mozilla/experimenter/issues/6129#issuecomment-893778432) in issue #6129.

But, things have been updated since then. [Here's the Jetstream change that corresponds to this type](https://github.com/mozilla/jetstream/commit/530469c5facf9f1573a3882d831e1dd73c6cb3eb).

And here's what the data in a prod experiment looks like from the analysis endpoint

![Capture](https://user-images.githubusercontent.com/21687/130279063-09af836d-1c01-41ef-82ea-6a9cb6140170.PNG)

LZoog replied:
Ack, yeah, I can see why that's confusing. We originally thought this was a problem with only the `reference_branch` and it was exposed in the metadata as only `reference_branch` (jetstream config schema version 3). Then, a few days later, #6200 was filed, which prompted changing this to that `external_config` object (schema version 4).

The metadata for that experiment was re-uploaded manually end of day yesterday by Anna, as well as a couple of others that had overrides, and this morning I went to double check that `external_config` was coming back on `metadata` on the experiment in that bug and we discovered we needed to "force jetstream data fetch" for that experiment since old completed experiments apparently keep that cache around for 3 days or so (I'm making note of this in a doc update).

lmorchard reviewed (approved):
LGTM!

LZoog commented:
I mentioned in my original PR comment that I wanted to check this out locally first for that experiment mentioned in the bug before merging - I'm having trouble doing that as well as even pulling up results for `testing-nimbus` locally per our docs (looks like `testing-nimbus` doesn't exist in production and I thhink it has to?). I'll merge this Monday morning after more investigating (AKA pinging someone that'll have an answer 😄) and will have a PR up to update the docs too.
